syne_tune/tuner_callback.py

```diff
@@ -192,44 +196,55 @@ def on_trial_result(self, trial: Trial, status: str, result: Dict, decision: str
 
         self._set_time_fields(result)
 
-        new_result = result[self.metric]
+        if self._target_metric is not None:
+
+            assert self._target_metric in result,  f"{self._target_metric} was not reported back to Syne tune"
+            new_result = self.metric_sign * result[self._target_metric]
 
-        if self.mode == 'max':
-            new_result *= -1
+            if self.curr_best_value is None or self.curr_best_value > new_result:
+                self.curr_best_value = new_result
+                self.curr_best_config = trial.config
+                self.writer.add_scalar(self._target_metric, result[self._target_metric], self._iter)
 
-        if self.curr_best_value is None or self.curr_best_value > new_result:
-            self.curr_best_value = new_result
-            self.curr_best_config = trial.config
-            self.writer.add_scalar(self.metric, result[self.metric], self._iter)
+            else:
+                opt = self.metric_sign * self.curr_best_value
+                self.writer.add_scalar(self._target_metric, opt, self._iter)
 
-        else:
-            opt = self.curr_best_value
-            if self.mode == 'max':
-                opt *= -1
-            self.writer.add_scalar(self.metric, opt, self._iter)
+            for key, value in self.curr_best_config.items():
+                self.writer.add_scalar(f'optimal_{key}', value, self._iter)
 
-        for key, value in self.curr_best_config.items():
-            self.writer.add_scalar(f'optimal_{key}', value, self._iter)
+        for metric in result:
+            if self.ignore_metrics is not None and metric not in self.ignore_metrics:
```

Comment:
Nit: It might be cleaner to set `ignore_metrics` to `{}` when nothing is passed and then directly do:
```suggestion
            if metric not in self.ignore_metrics:
```